packages/devtools_app/lib/src/shared/primitives/utils.dart

```diff
@@ -1061,9 +1061,20 @@ extension StringExtension on String {
     return true;
   }
 
-  /// Whether [other] is a case insensitive match for this String
-  bool caseInsensitiveEquals(String? other) {
-    return toLowerCase() == other?.toLowerCase();
+  /// Whether [other] is a case insensitive match for this String.
+  ///
+  /// If [pattern] is a [RegExp], this method will return true iff this String
+  /// is a complete [RegExp] match, meaning that the regular expression finds a
+  /// match with starting index 0 and ending index [this.length].
+  bool caseInsensitiveEquals(Pattern? pattern) {
+    if (pattern is RegExp) {
+      assert(!pattern.isCaseSensitive);
```

Comment:
What happens when this assert is  triggered?

Reply:
Right now this would just fail silently but we could catch it in development. This is what we have in caseInsensitiveContains too. Maybe we should provide some validation in the search / filter text fields or add a note when reg exp is enabled.